x-pack/plugins/cases/public/components/visualizations/actions/add_to_existing_case.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  hasCasePermissions,
+  isLensEmbeddable,
+  hasInput,
+  getCaseOwner,
+  getCasePermissions,
+} from './utils';
+import { KibanaContextProvider } from '../../../common/lib/kibana';
+
+import CasesProvider from '../../cases_context';
+import type { ActionContext, CaseUIActionProps, DashboardVisualizationEmbeddable } from './types';
+import { useCasesAddToExistingCaseModal } from '../../all_cases/selector_modal/use_cases_add_to_existing_case_modal';
+import { ADD_TO_EXISTING_CASE_DISPLAYNAME } from './translations';
+
+export const ACTION_ID = 'embeddable_addToExistingCase';
+export const DEFAULT_DARK_MODE = 'theme:darkMode' as const;
+
+interface Props {
+  embeddable: DashboardVisualizationEmbeddable;
+  onSuccess: () => void;
+  onClose: (theCase?: CaseUI) => void;
+}
+
+const AddExistingCaseModalWrapper: React.FC<Props> = ({ embeddable, onClose, onSuccess }) => {
+  const modal = useCasesAddToExistingCaseModal({
+    onClose,
+    onSuccess,
+  });
+
+  const attachments = useMemo(() => {
+    const { attributes, timeRange } = embeddable.getInput();
+
+    return [
+      {
+        comment: `!{lens${JSON.stringify({
+          timeRange,
+          attributes,
+        })}}`,
+        type: CommentType.user as const,
+      },
+    ];
+  }, [embeddable]);
+  useEffect(() => {
+    modal.open({ getAttachments: () => attachments });
+  }, [attachments, modal]);
+
+  return null;
+};
+
+AddExistingCaseModalWrapper.displayName = 'AddExistingCaseModalWrapper';
+
+export const createAddToExistingCaseLensAction = ({
+  core,
+  plugins,
+  storage,
+  history,
+  caseContextProps,
+}: CaseUIActionProps) => {
+  const { application: applicationService, theme, uiSettings } = core;
+  let currentAppId: string | undefined;
+  applicationService?.currentAppId$.subscribe((appId) => {
+    currentAppId = appId;
+  });
+
+  const owner = getCaseOwner(currentAppId);
+  const casePermissions = getCasePermissions(applicationService.capabilities, owner);
+
+  return createAction<ActionContext>({
+    id: ACTION_ID,
+    type: 'actionButton',
+    getIconType: () => 'casesApp',
+    getDisplayName: () => ADD_TO_EXISTING_CASE_DISPLAYNAME,
+    isCompatible: async ({ embeddable }) =>
+      !isErrorEmbeddable(embeddable) &&
+      isLensEmbeddable(embeddable) &&
+      hasCasePermissions(casePermissions) &&
+      hasInput(embeddable),
+    execute: async ({ embeddable }) => {
+      const targetDomElement = document.createElement('div');
+
+      const cleanupDom = (shouldCleanup?: boolean) => {
+        if (targetDomElement != null && shouldCleanup) {
+          unmountComponentAtNode(targetDomElement);
+        }
+      };
+
+      const onClose = (theCase?: CaseUI, isCreateCase?: boolean) => {
+        const closeModalClickedScenario = theCase == null && !isCreateCase;
+        const caseSelectedScenario = theCase != null;
+        const shouldCleanup = closeModalClickedScenario || caseSelectedScenario;
+        cleanupDom(shouldCleanup);
```

Comment:
If I'm understanding this correctly we don't clean up if a case is created. Just curious why that is?

Reply:
Yeah, as after clicking creating a case from the modal, we close the modal and open the flyout. If we clean up dom when closing the modal, then the flyout won't open.